### PR TITLE
Added Type hints for XLM TF

### PR DESCRIPTION
**Pull request by IMvision12**
Based on Issue #16059

I have added type hints for Tensorflow XLM model.

@Rocketknight1 Could you kindly check if this is fine?

### CODE REVIEW

**Review by Rocketknight1 — CHANGES_REQUESTED** [review]
Hi, thanks for the PR! It looks mostly good, but there are a couple of issues:

1. Be careful not to overwrite the default value for `training`! Also, `training` cannot be `None`, so it shouldn't be `Optional`. Instead, those lines should be `training: bool = False`.
2. Be careful with some of the types! `output_attentions` and `output_hidden_states` should be `Optional[bool]`, while `start_positions` and `end_positions` are label tensors for QA models, so they should be `Optional[Union[np.ndarray, tf.Tensor]]`

Once those are fixed, though, this looks ready to be merged, so thank you! You can ping me again whenever it's ready.

**Comment by IMvision12** [comment]
@Rocketknight1 Thanks for the feedback, I have updated the files.

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 1]
```diff
@@ -844,19 +845,19 @@ def prepare_inputs_for_generation(self, inputs, **kwargs):
...
-        output_hidden_states=None,
-        return_dict=None,
-        training=False,
+        input_ids: Optional[TFModelInputType] = None,
+        attention_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        langs: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        token_type_ids: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        position_ids: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        lengths: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        cache: Optional[Union[np.ndarray, tf.Tensor]] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:854]:
```suggestion
        cache: Optional[Dict[str, tf.Tensor]] = None,
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 2]
```diff
@@ -916,20 +917,20 @@ def __init__(self, config, *inputs, **kwargs):
...
-        return_dict=None,
-        labels=None,
-        training=False,
+        input_ids: Optional[TFModelInputType] = None,
+        attention_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        langs: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        token_type_ids: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        position_ids: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        lengths: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        cache: Optional[Union[np.ndarray, tf.Tensor]] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:926]:
```suggestion
        cache: Optional[Dict[str, tf.Tensor]] = None,
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 3]
```diff
@@ -1023,20 +1024,20 @@ def dummy_inputs(self):
...
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:1033]:
```suggestion
        cache: Optional[Dict[str, tf.Tensor]] = None,
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 4]
```diff
@@ -1147,20 +1148,20 @@ def __init__(self, config, *inputs, **kwargs):
...
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:1157]:
```suggestion
        cache: Optional[Dict[str, tf.Tensor]] = None,
```

File: src/transformers/models/xlm/modeling_tf_xlm.py [hunk 5]
```diff
@@ -1232,21 +1233,21 @@ def __init__(self, config, *inputs, **kwargs):
...
-        start_positions=None,
-        end_positions=None,
-        training=False,
+        input_ids: Optional[TFModelInputType] = None,
+        attention_mask: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        langs: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        token_type_ids: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        position_ids: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        lengths: Optional[Union[np.ndarray, tf.Tensor]] = None,
+        cache: Optional[Union[np.ndarray, tf.Tensor]] = None,
```
Comment (Rocketknight1) [inline src/transformers/models/xlm/modeling_tf_xlm.py:1242]:
```suggestion
        cache: Optional[Dict[str, tf.Tensor]] = None,
```

**Review by Rocketknight1 — APPROVED** [review]
LGTM! I also just realized the cache actually has a weird type too, but I can set that here and then merge once the tests pass. Thank you!
